core/src/test/scala/org/broadinstitute/dsde/rawls/googleProject/GoogleProjectRegistrationServiceSpec.scala

```diff
@@ -0,0 +1,529 @@
+package org.broadinstitute.dsde.rawls.googleProject
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import com.google.api.services.cloudbilling.model.ProjectBillingInfo
+import org.broadinstitute.dsde.rawls.billing.BillingRepository
+import org.broadinstitute.dsde.rawls.{RawlsExceptionWithErrorReport, TestExecutionContext}
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, SamDAO, SlickDataSource}
+import org.broadinstitute.dsde.rawls.mock.RemoteServicesMockServer
+import org.broadinstitute.dsde.rawls.model.{
+  CreationStatuses,
+  GoogleProjectId,
+  GoogleProjectRegistration,
+  RawlsBillingAccountName,
+  RawlsBillingProject,
+  RawlsBillingProjectName,
+  RawlsRequestContext,
+  SamBillingProjectActions,
+  SamGoogleProjectActions,
+  SamResourceAction,
+  SamResourceTypeName,
+  SamResourceTypeNames
+}
+import org.mockito.Mockito.{verify, when}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.broadinstitute.dsde.rawls.util.MockitoTestUtils
+import org.mockito.ArgumentCaptor
+import org.scalatest.matchers.should.Matchers
+import org.mockito.ArgumentMatchers.{eq => mockitoEq, _}
+
+import java.util.UUID
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+
+class GoogleProjectRegistrationServiceSpec
```

Comment:
Might be worth adding a test for what happens when updating the billing account with Google fails for whatever reason